lib/ansible/plugins/inventory/linode.py

```diff
@@ -200,9 +224,9 @@ def parse(self, inventory, loader, path, cache=True):
         self._get_instances_inventory()
 
         config_data = self._read_config_data(path)
-        regions, types = self._get_query_options(config_data)
+        regions, types, use_public_ip = self._get_user_options(config_data)
         self._filter_by_config(regions, types)
 
         self._add_groups()
-        self._add_instances_to_groups()
-        self._add_hostvars_for_instances()
+        self._add_instances_to_groups(use_public_ip)
```

Comment:
Thinking ahead, in case there are other ways to specify the identifier for an instance (lable, ip, etc.), we might implement here a `self._determine_host_identifier` or so, which negotiates the options and decides if we're using the `public_ip` or the `label` and then passes that to the `_add_instances_to_groups` and `_add_hostvars_for_instances`? We could then eliminate the `if/else` clause in those function sand keep the control flow more simple. WDYT?